openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -83,10 +92,24 @@ def offer_html(self):
 
     @property
     def content_type_gating_enabled(self):
-        return ContentTypeGatingConfig.enabled_for_enrollment(
-            user=self.effective_user,
-            course_key=self.course_key,
-        )
+        course_key = self.course_key
+        user = self.effective_user
+        if is_masquerading_as_staff(user, course_key):
```

Comment:
We _could_ also short-circuit sooner if they're not masquerading _at_ _all_, but these should be lightweight...